platform/linux-generic/odp_schedule_basic.c

```diff
@@ -1430,15 +1430,18 @@ static inline int do_schedule(odp_queue_t *out_q, odp_event_t out_ev[], uint32_t
 		sched_local.grp_epoch = epoch;
 	}
 
-	grp_id = sched_local.grp_weight[grp_round];
+	first_id = sched_local.grp_weight[grp_round];
 
-	/* Schedule queues per group and priority */
-	for (i = 0; i < num_grp; i++) {
-		int grp, prio;
+	for (prio = 0; prio < NUM_PRIO; prio++) {
+		grp_id = first_id;
+
+		for (i = 0; i < num_grp; i++) {
```

Comment:
Now that the order of iteration of priorities and groups has changed, I wonder if it would make sense to change the layout of the prio_q_mask and prio_q arrays to better match the order of access by making priority the outer index and group the inner index.

Reply:
I think [group][prio] should be still more efficient. Typically, only a small number of groups are used or a thread is part of only some groups (e.g. 0, 1 and 13). When a group is active for a thread, scheduler needs to check all 8 priority levels of it. This way array packs per group information closer together in memory.

Reply:
Maybe so. Hard to say which is best. But changing the layout does not appear to be a small change so even if it made sense, it probably should be done separately.